config/routes.rb

```diff
@@ -19,10 +19,15 @@
   # users
   resources :users, only: [:new, :create, :edit, :update, :show]
 
-  # folders & download
+
   resources :folders, path: :f, only: [:show]
   resources :folders, path: "f/:id", only: [:new, :create]
-
+  resources :folders, path: :f, only: [:show] do
+    get "/share", to: "folders/collaborations#new"
+    post "/share", to: "folders/collaborations#create"
```

Comment:
I like these routes! For the purposes of simplicity, we should refactor/update our routes so we're always using `:folder_id` instead of mixing between `:folder_id` and `:id`.